content/posts/why-meson/index.md

```diff
@@ -0,0 +1,248 @@
+last few months this problem was caused by wrong dependencies of `sdb`, in
+the past due to wrong dependencies of the `capstone` target).
+- `meson` can run everywhere `python3` can. This includes a very wide range
+of platforms nowadays. It automatically provides a very powerful scripting
+language, python, that you are guaranteed to find on the build machine.
+Moreover, it can be used with various backends, like Ninja, Visual Studio and
+Xcode, which means it can be used to generate a Visual Studio solution that
+you can import there.
+- It forces you to build out-of-source, meaning that no changes (mostly) will
+be done to your source directory, which must contain only the source files of
+your project and not be mixed with other auto-generated files like
+executables or object files. This also allows you to have the project
+compiled with different options or with slightly different code, cleanly
+separated in different directories.
+- Due to its declarative nature, it does not matter whether a dependency is
+in a path or another or if it comes from the system or it was bundled with
+the source code. You just define `capstone_dep` variable properly in one of
+your `[meson.build](http://meson.build)` files and you reference it wherever
+it is needed, leaving all the details to `meson` itself. This encourages
+splitting the repository in sub-projects when it makes sense, in contrast
+with the ACR/Make system where even a small change to e.g. SDB path would
+require rewriting several Makefiles. If in the future some systems will ship
+their own version of SDB, we would just need to change few lines in the
+definition of `sdb_dep` to actually take the system library instead of the
+bundled one and no other place would need to be changed to make sure
+everything is compiled/linked with the right headers/libraries.
+- In case of problems with `meson` there is a healthy community out there
+ready to help you, a nice and extensive documentation and active developers
+that improve the system with new releases. New developers who want to work on
+our build system can easily find other examples online and have available
+documentation to get them up to speed.
+- Many complex low-level pure C projects recently switched to Meson:
+[Mesa](https://mesa3d.org/),
+[Wayland](https://github.com/wayland-project/wayland),
+[PipeWire](https://github.com/PipeWire/pipewire),
+[QEMU](https://www.qemu.org/), and [many
+others](https://mesonbuild.com/Users.html). We are not alone in this!
+
+# Examples of using meson
+
+## Development process
+
+As a developer when you download Rizin, you can install it for your user in
+`~/.local`, so you don't need root access to install files. You can do this
+with `meson --prefix=~/.local build; ninja -C build install`. After that, you
+can change the source code however you need and then run `ninja` again with
+`ninja -C build`. Only the changed files are re-built.
```

Comment:
Well, it seems the answer to my above question is here :)
So maybe fix the first mention (it's not `meson` per se but `ninja`), however you like it.